Makefile.setup

```diff
@@ -39,17 +42,39 @@ endif
 help::
 	echo "Available setup targets:"
 	echo "  setup                 to perform Git repository setup"
+	echo "  setup-conan           to install Conan profile"
 	echo "  install-system-deps   to install build (and development) system requirements"
 	echo "  install-python-deps   to install Python runtime requirements with ${PIP}"
 	echo "  install-sphinx-deps   to install Sphinx runtime requirements with ${PIP}"
 	echo "  install-python-tools  to install Python development tools with ${PIPX}"
 	echo
 
-.PHONY: setup
+.PHONY: setup setup-conan
 setup:
 	git config --local core.hooksPath .git-hooks/
 
-.PHONY: install-system-deps install-ubuntu-deps install-archlinux-deps install-python-deps install-python-tools
+CONAN_DIR=$(or $(shell conan config home), ${HOME}/.conan)
+setup-conan:
+	# Install Conan if it is not available.
+	if ! command -v conan >/dev/null 2>&1; then ${PIP} install --user conan; fi
+	# Initialize Conan configuration if it doesn't already exist.
+	if [ ! -d "${CONAN_DIR}" ]; then \
+		conan config init; \
+		conan config set general.revisions_enabled=1; \
```

Comment:
The implications of activating package revision mode are not entirely clear to me. Let's discuss..

Reply:
Just realized that `${CONAN_DIR}` will be created when doing `$ conan config home`. So for which situations is this code block meant?

Reply:
...

Reply:
I expected `conan config home` to be a read-only operation. Guess I was wrong.

Reply:
> The implications of activating package revision mode are not entirely clear to me. Let's discuss..

I read up on it in the Conan documentation and it doesn't look like it should affect those much who don't also have it activated. It basically checksums everything explicitely part of the recipe and calculates a revision from that. It does this via the conan manifest. Usually the only file that is part of that is the recipe itself. But it can also include a CMakeLists.txt file that we add or patches and the like.

I don't see any downsides to using it; in fact this should solve some issues we've been having when uploading packages to the Artifactory.

Reply:
I addition, it will become the default in Conan 2.0 so we might as well start now.

Reply:
> Just realized that `${CONAN_DIR}` will be created when doing `$ conan config home`. So for which situations is this code block meant?

How do you figure this actually? I looked at the documentation and the code in Conan and it doesn't look like it should create the directory. Is this a Conan bug?

Reply:
Ok, after trying some things out, it looks like the first time you run conan, regardless of what you execute - even if it's only `conan --help`, it will go and create the directory. This is... unexpected.

Reply:
I reworked the setup-conan target to take this into account though.